odiglet/cmd/main.go

```diff
@@ -85,19 +83,27 @@ func main() {
 		os.Exit(-1)
 	}
 
+	instrumentationManager := ebpf.NewManager(
+		mgr.GetClient(),
+		log.Logger,
+		map[ebpf.FactoryID]ebpf.Factory{
+			ebpf.FactoryID{
+				Language: common.GoProgrammingLanguage,
+				OtelSdk:  common.OtelSdkEbpfCommunity,
+			}: sdks.NewGoInstrumentationFactory(),
+		},
+		procEvents,
+	)
+
+	go instrumentationManager.Run(ctx)
```

Comment:
What is the motivation behind this?

Reply:
So that we don't start to run the `instrumentationManager` which uses the `mgr` before we finished initializing the controller runtime manager?

Not sure if it will make any different, but I think it's nice to have all setup/initialization code, then run all components

Reply:
I agree that it will make the code more readable, but I'm not sure if it has an impact on correctness, if we start the instrumentation manager too early? it means some of the events will be idle - but it can still get process events and instrument stuff before the manager starts.
I also think we can refactor the main function of Odiglet to a structure like you suggested, but prefate to do it in a separate PR.
WDYT?